backends/graphics/windowed.h

```diff
@@ -94,8 +94,11 @@ class WindowedGraphicsManager : virtual public GraphicsManager {
 			error("convertVirtualToWindow called without a valid draw rect");
 		}
 
-		return Common::Point(targetX + (x * targetWidth + sourceWidth / 2) / sourceWidth,
-		                     targetY + (y * targetHeight + sourceHeight / 2) / sourceHeight);
+		int windowX = targetX + (x * targetWidth + sourceWidth / 2) / sourceWidth;
+		int windowY = targetY + (y * targetHeight + sourceHeight / 2) / sourceHeight;
+
+		return Common::Point(CLIP<int>(windowX, 0, targetWidth - 1),
```

Comment:
There is a catch here: it should be clipped between `targetX` and `targetX + targetWidth - 1` ;-)
And the same thing on the next line as well.